src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SqlServerType.cs

```diff
@@ -90,7 +91,14 @@ public static SqlServerType CalculateServerType(ServerInfo serverInfo)
             }
             else if (serverInfo.IsCloud)
             {
-                serverType = SqlServerType.AzureV12;
+                if (serverInfo.EngineEditionId == 6 && serverVersion.StartsWith("12", StringComparison.Ordinal))
```

Comment:
Oh that's fun...we only know about versions up to 2017 in here? 🤦‍♂️ (not related to your PR)